src/compiler/semantic/mod.rs

```diff
@@ -286,6 +290,68 @@ impl SymTable {
                 .join("/")
         }
     }
+
+    pub fn find_symbol_by_offset(&self, filename: String, offset: usize) -> Option<SymTableEntry> {
+        let mut symbols_by_proximity = BTreeMap::<i32, SymTableEntry>::new();
+
+        for scope in self.scopes.values() {
+            for (_, entry) in &scope.symbols {
+                // If the entry is not in the same file, check its usages
+                if entry.definition_ref.get_filename() != filename.clone() {
+                    SymTable::look_in_usages(
+                        entry,
+                        filename.clone(),
+                        offset,
+                        &mut symbols_by_proximity,
+                    );
+                } else {
+                    let proximity = entry.definition_ref.proximity_score(offset);
+                    // If the current entry is not enclosing the offset, check the usages of that
+                    // entry
+                    if proximity == -1 {
+                        SymTable::look_in_usages(
+                            entry,
+                            filename.clone(),
+                            offset,
+                            &mut symbols_by_proximity,
+                        );
+                    // If the current entry is enclosing the offset, add it to the map
+                    } else {
+                        symbols_by_proximity.insert(proximity, entry.clone());
+                    }
+                }
+            }
+        }
+
+        if symbols_by_proximity.is_empty() {
+            return None;
+        } else {
+            // Return the first symbol in the map because BTreeMap is sorted by the key (which is
+            // the proximity in our case)
+            return symbols_by_proximity
+                .iter()
+                .next()
+                .map(|(_, entry)| entry.clone());
+        }
+    }
+
+    fn look_in_usages(
```

Comment:
@alxkzmn it seems this method doesn't use `&self` but `&SymTableEntry` perhaps should be a method in `SymTableEntry`